rfd/0225-teleport-extended-support.md

```diff
@@ -0,0 +1,146 @@
+things: stability and extended support. We can achieve these without introducing a Teleport LTS
+release by changing our process slightly.
+
+We can reduce the frequency of Teleport major releases from every ~4 months to once a year and cut
+down the number of supported releases from N-2 to N-1. That would extend the support window for major
+releases from ~12 months to ~24 months at the expense of supporting fewer major versions. Reducing
+the number of versions also means it takes fewer upgrade cycles to get to the most recent version of Teleport.
+
+This satisfies customer desire for major releases to be supported for a longer period of time
+without impacting our current upgrade process or component compatibility guarantees. However, it
+comes at the expense of reducing the frequency at which we can make breaking changes.
+
+As dictated by semantic versioning, a major release should be reserved for making any backward-incompatible
+changes to the public API. We've often relied on this four month release cycle to quickly iterate
+on APIs that don't work or cause issues for one reason or another. With yearly major releases this
+becomes a less viable solution and a stronger emphasis must be put both on backward and forward compatibility. 
+
+Going forward, solutions to problems should attempt to preserve compatibility at all costs and
+rely less on all customers adhering to our upgrade procedure. Instead of relying on versions of
+clients, and that the Auth server will reject outdated clients we should put a stronger emphasis
+on negotiating compatibility. If clients and servers in our control can inform each other about
+what features they support in some manner then both can mutually agree on which features and APIs
+can be safely used.
+
+To address the stability concerns, all new features *MUST* only be added to the most recent major version.
+The previous major version goes into maintenance mode the moment a new major version is released.
+In other words, even though we will now support two major releases, N and N-1, new features *MUST* only
+be included in version N. N-1 will continue to receive bug fixes and security patches, but not new features.
+
+### Toolchain Implications
+
+Extending support of a Teleport means that it may outlive releases of toolchains used to build Teleport.
+Go releases are supported for ~12 months. They release a new major version every ~6 months, and are EOL
+when two newer major releases exist.
+
+While Go is typically good about preserving compatibility, new releases do occasionally alter behavior
+that require changes to Teleport. We could align our release cycle to the Go release cadence to reduce
+the number of times the toolchain would need to be upgraded but that doesn't buy us much.
+
+This isn't an entirely new problem as our current N-2 and N-1 release outlive Go toolchains today.
+We should continue to only bump the Go toolchain on branches _after_ it has gone through a test plan.
```

Comment:
after what has gone through a test plan? if we only do a major release once a year will we only be doing the test plan once a year? is the idea that we would upgrade the toolchain used to build v18 releases after we go through the v19 test plan using the latest toolchain?